src/directory.ts

```diff
@@ -24,18 +26,20 @@ export class Directory {
 					const secondExt = pathArray.pop() as string;
 					return (
 						FILE_EXTENSIONS.contract.includes(ext) ||
-						(ext === FILE_EXTENSIONS.js &&
-							FILE_EXTENSIONS.chain.includes(secondExt) &&
-							pathArray.pop()) ||
-						(ext === FILE_EXTENSIONS.json &&
-							FILE_EXTENSIONS.jsonDouble.includes(secondExt) &&
-							pathArray.pop()) ||
+						Boolean(
+							ext === FILE_EXTENSIONS.js &&
+								FILE_EXTENSIONS.chain.includes(secondExt) &&
+								pathArray.pop()
+						) ||
+						Boolean(
+							ext === FILE_EXTENSIONS.json &&
+								FILE_EXTENSIONS.jsonDouble.includes(secondExt) &&
+								pathArray.pop()
+						) ||
 						ext === FILE_EXTENSIONS.conf
 					);
 				},
```

Comment:
Better readable, still a little bit cryptic, and high cognitive load.
Maybe a small comment, with what output expected, could help the future you or dev ;p

Reply:
I do not totally get what you mean here, like adding just the type that keep and ignore should return like
```
keep: (path: string): boolean => {}
ignore: (path: string): boolean => {}
```
or to explain better at then end what file should being tracked by livedirectory with a comment like
```
This should keep track of file that ends up with:
* .[zen]
* .[chain].js
* .[keys|data|metdata|schema].json
* .conf
that are present in the zencodeDirectory and outside of the autorun folder
```
or both?


P.s. chain and zencode extension are array since in the future we may want to support also slang or other extensions